x/evm/keeper/keeper.go

```diff
@@ -82,14 +88,32 @@ func (k Keeper) CommittedCtx() sdk.Context {
 	return k.committedCtx
 }
 
+// Ctx returns the top of context stack
+func (k Keeper) Ctx() sdk.Context {
+	l := len(k.cachedContexts)
+	if l == 0 {
+		return k.committedCtx
+	}
+	return k.cachedContexts[l-1].ctx
+}
+
+// CommitCacheContexts commit and remove all the cached contexts
+func (k Keeper) CommitCacheContexts() {
+	// commit in reversed order
+	for i := len(k.cachedContexts) - 1; i >= 0; i-- {
+		k.cachedContexts[i].commit()
+	}
+	k.cachedContexts = []cachedContext{}
+}
+
 // Logger returns a module-specific logger.
 func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", types.ModuleName)
 }
 
 // WithContext sets an updated SDK context to the keeper
 func (k *Keeper) WithContext(ctx sdk.Context) {
```

Comment:
change name to SetContext , because it's modifying function

Reply:
hmm, @fedekunze , what do you think?

Reply:
out of scope, we can decide and change it later